lib/auth/join.go

```diff
@@ -380,6 +368,22 @@ func (a *Server) generateCertsBot(
 			RemoteAddr: req.RemoteAddr,
 		},
 	}
+
+	fingerprint := sha256.Sum256(req.PublicTLSKey)
```

Comment:
I know we specified a "sha256 fingerprint" in the RFD but I'm not actually entirely clear on exactly what that is. I've just hashed and hex encoded it here.

Reply:
TBH - The hash/fingerprint mostly existed for the purposes of identifying a BotInstance when the initial plan wasn't to use UUIDs - we could probably just drop this field now.

If you want to keep it, there's not a whole lot of agreement on fingerprinting/hashing public keys. In other places of Teleport, e.g the TPM authentication, I achieved this by hashing the PKIX encoded public key. However, it looks like our Join endpoints support the public key being submitted in either PKIX or PKCS1 encoding - if we are going to have a hash, we should de-encode from PKIX or PKCS1 and re-encode always into one specific encoding before hashing to ensure that the same public key receives the same fingerprint regardless of encoding.

Reply:
Fair enough, I'll drop the field. I think it'd be a tricky identifier regardless - easier than the public key, sure, but still a difficult thing for end users to look at and compare vs the UUID.